src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/WindowsPrincipal.cs

```diff
@@ -161,7 +162,7 @@ public virtual bool IsInRole(SecurityIdentifier sid)
             if (!Interop.Advapi32.CheckTokenMembership((_identity.ImpersonationLevel != TokenImpersonationLevel.None ? _identity.AccessToken : token),
                                                   sid.BinaryForm,
                                                   ref isMember))
-                throw new SecurityException(new Win32Exception().Message);
+                throw new SecurityException(Marshal.GetPInvokeErrorMessage(Marshal.GetLastPInvokeError()));
```

Comment:
Hmm, as a test of the new API, this is an improvement, but not a great one. It is now self explanatory, but it's verbose.

Something like `Marshal.GetPinvokeErrorMessage()` (ie a parameterless overload) would improve this. But I believe we discussed this and the concern was that it is less clear that one must call this immediately after the pinvoke since it captures last error. It would be easy to make this call later, at the point where you need the message, yet potentially the error code has changed.

Reply:
I would be inclined to suggest that API review consider this problem. I suspect `Marshal.GetPinvokeErrorMessage()`  might be good.

Reply:
A name like `Marshal.GetLastPInvokeErrorMessage()` might emphasize the importance of calling this immediately.